File: python/day2/day2_pt2.py
passwords = []
valid_passwords = 0


with open('input.txt') as f:
	for line in f:
		passwords.append(line.strip())

for password in passwords:
	context = password.split()
	
	character = context[1].replace(':', '')
	plain_password = context[2]
	positions = context[0].split('-')
	splitted_pass = list(plain_password)
	
	verification = []
	if splitted_pass[int(positions[0]) -1] == character:
		verification.append(True)
	if splitted_pass[int(positions[1]) -1] == character:
		verification.append(True)
	
	if verification.count(True) == 1:
		print('{} is valid'.format(password))
		valid_passwords += 1
	else:
		print('{} is not valid'.format(password))

print(valid_passwords)

File: clojure/day2/day2_pt2.clj
(ns advent-of-code.day2.day2-pt2
  (:require [clojure.string :as string]
            [clojure.set :as s]
            [clojure.edn :as edn]))

(def input-data
  (->> (slurp "/tmp/input.txt")
       string/split-lines))

(defn parse-data [data]
  (let [parsed-data (string/split data #" ")]
    {:indexes       (-> (nth parsed-data 0)
                      (string/split #"-"))
     :character   (first (concat (string/replace (nth parsed-data 1) #":" "")))
     :password-as-list (seq (nth parsed-data 2))
     }))

(defn verify-position [coll]
  (let [int-range (map edn/read-string (:indexes coll))
        actual-range (map dec int-range)
        password-list (:password-as-list coll)
        password-positions [(nth password-list(first actual-range)) (nth password-list (last actual-range))]]
    (if (not= true (apply = password-positions))
      (some #{(:character coll)} password-positions)
      nil)))

(->> input-data
     (map parse-data)
     (map verify-position)
     (filter #(not= nil %))
     count)
